mage_ai/tests/data_cleaner/transformer_actions/test_custom_actions.py

```diff
@@ -0,0 +1,189 @@
+from mage_ai.data_cleaner.transformer_actions.custom_action import execute_custom_action
+from mage_ai.tests.base_test import TestCase
+from pandas.testing import assert_frame_equal
+import pandas as pd
+import numpy as np
+
+
+class CustomActionTests(TestCase):
+    def test_decorated_custom_action(self):
+        df = pd.DataFrame(
+            [
+                ['1', 1.000, '2021-10-01', 'Store 1', 23023],
+                ['1', None, '2021-10-01', 'Store 2', np.nan],
+                [np.nan, 1100, '', '', 90233],
+                ['2', None, None, 'Store 1', 23920],
+                ['2', 12.00, '2021-09-01', None, np.nan],
+                ['2', 125.0, '2021-09-01', 'Store 3', 49833],
+            ],
+            columns=[
+                'group_id',
+                'price',
+                'group_churned_at',
+                'store',
+                'zip_code',
+            ],
+        )
+        expected_df = pd.DataFrame(
+            [
+                [1.0, 1.000, '2021-10-01', 'Store 1', 23023, 1],
+                [1.0, None, '2021-10-01', 'Store 2', np.nan, 1],
+                [np.nan, 1100, np.nan, np.nan, 90233, None],
+                [2.0, None, None, 'Store 1', 23920, 2],
+                [2.0, 12.00, '2021-09-01', None, np.nan, 2],
+                [2.0, 125.0, '2021-09-01', 'Store 3', 49833, 2],
+            ],
+            columns=['group_id', 'price', 'group_churned_at', 'store', 'zip_code', 'new_col'],
+        )
+        action = {
+            'action_type': 'custom',
+            'axis': 'row',
+            'action_code': 'from mage_ai.data_cleaner.column_type_detector import NUMBER,'
```

Comment:
Let's write multiline text in the following format
''''
from mage_ai....
....
....
'''

Reply:
Fixed (albeit with strange formatting in order to adjust for indent)